smart-contracts/contracts/mixins/MixinKeys.sol

```diff
@@ -173,6 +173,36 @@ contract MixinKeys is
     return key.tokenId;
   }
 
+  function _extendKey(
+    address _recipient
+  ) internal 
+    returns (
+      uint newTimeStamp
+    )
+  {
+    Key memory key = getKeyByOwner(_recipient);
+
+    if (key.expirationTimestamp > block.timestamp) {
```

Comment:
Let's actually make it so that it is possible to extend keys even if they are not expired, but "almost" expired (i.e within 5% of their expiration)